app.py

```diff
@@ -493,10 +499,92 @@ def course(course_id=None):
+                                                                 d, 
+                                                                 request.form['aType'])
+                            flash(response_tuple[1])
+                            return redirect(url_for('course', course_id=course_id))
+                        elif request.form['submit'] == 'delete':
+                            if assignmentdb.exists(course_id, 
+                                                   request.form['delete_name']):
+                                assignmentdb.remove(course_id,
+                                                    request.form['delete_name'])
+                                flash("Successfully removed assignment")
+                            else:
+                                flash("Error: Assignment not found")
+                            return redirect(url_for('course', course_id=course_id))
+                        else:
+                            flash("Invalid request")
+                            return redirect(url_for('course', course_id=course_id))
+                    else:
+                        flash(response_tuple[1])
+                        return redirect(url_for('course', course_id=course_id))
+                else:
+                    flash("Invalid request")
+                    return redirect(url_for('course', course_id=course_id))                             
+                           
 
     return redirect(url_for('index'))
 
+@app.route('/assignment/<course_id>/<name>', methods=['GET','POST'])
+@redirect_if_not_logged_in
+def assignment(course_id=None, name=None):
+    if course_id and name:
+        if request.method == 'GET':
+            hasPermissionToView = False
+            if session['userType'] == 'student':
+                hasPermissionToView = studentdb.hasCourse(session['email'],
+                                                          course_id)
+            elif session['userType'] == 'teacher':
+                hasPermissionToView = teacherdb.hasCourse(session['email'],
+                                                          course_id)
+            course = coursedb.get(course_id)
+            if course and course.count() == 1:
+                course = course[0]
+                teacher = teacherdb.get('', teacher_id=course['teacherId'])
+                if teacher and teacher.count() == 1:
+                    teacher = teacher[0]['name']
+                    students = []
+                    assignment = assignmentdb.getByCourse(course_id)
+                    if hasPermissionToView:
+                        for studentId in course['students']:
+                            student = studentdb.get('', student_id=studentId)
+                            if student and student.count() == 1:
+                                students.append(student[0])
+            return render_template('assignment.html', course_data=course, 
```

Comment:
Your return statement needs to be farther inside the if block; the teacher and students variables are not yet declared/defined